dev-guide/test_framework.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: index
+title: GCAM Testing Framework
+gcam-version: v5.2
+devguide: dev-guide.html
+---
+## GCAM Testing Framework
+This page is here to serve as guidance for GCAM developers who may need to update or extend the GCAM Testing Framework.  Note the GCAM testing framework is meant to automate many of the steps developers must do to verify their changes do not break GCAM on any of the platforms GCAM is known to run on.  The services we use to do this are provided internal to PNNL, as noted below, therefore can not be used outside of the network.
+
+### Notes on initializing and updating the testing framework
+Most users will interact with the testing framework only through the pull request interface and will not need to worry about updating the testing framework.  Some instances of when they would include:
+* Updating any of the third party library dependencies (excluding gcamdata as that is self contained in the `DESCRIPTION` file)
+* They need to extend or add tests
+* A change that affects the way we run Core scenarios such as how we run a policy
+
+As described in the details [below](test_framework.html#background) the testing framework is contained in it's own Git repository but included in the GCAM core repository as a submodule.  Thus to get the scripts you will need to update it:
+```
+> cd <GCAM workspace>
+> git submodule update --init util/testing-framework
+```
+
+To understand how the testing framework is organized and how it all fits together they can read the detailed description [below](test_framework.html#background).
+
+The development cycle is the same as any other:
+```
+> cd util/testing-framework
+> git checkout -b my-new-feature-branch
+> # make the required changes
+> git add .
+> git commit
+> # don't forget to push your changes!
+> git push
+```
+
+However where it differs is we now need to update the submodule "pointer" in the gcam workspace as well:
+```
+> cd ../../
+> git add util/testing-framework
+> git commit
+> git push
+```
+
+At this point you can open your pull request and the automated tests will use your updated testing framework automatically.  In addition no other pull requests will yet be affected!  At this point you should also open a pull request in the [gcam-testing-framework repository](https://stash.pnnl.gov/scm/jgcri/gcam-testing-framework.git) and link to it from your GCAM pull request so all changes can be considered together.
+
+### Background
+Given this proposal will touch on some software / services which reviewers may not be familiar with it may be useful to review them and describe their purpose.  Note we are relying on these services being provided by PNNL's Dev Central which is currently being provided free of charge code.  It does of course imply these tests can only be launched from inside the PNNL network.
```

Comment:
STRUCTURE:  "*The following may* be useful to..." replacing "Given this proposal will touch on some software / services which reviewers may not be familiar with it..." since this is no longer a proposal.